src/main/java/hudson/plugins/pmd/PmdPublisher.java

```diff
@@ -129,17 +143,25 @@ public String getPattern() {
         return pattern;
     }
 
+   /**
+    * Sets the Ant file-set pattern of files to work with.
+    */
+    @DataBoundSetter
+    public void setPattern(final String pattern) {
+        this.pattern = pattern;
+    }
+
     @Override
     public Action getProjectAction(final AbstractProject<?, ?> project) {
         return new PmdProjectAction(project);
     }
 
     @Override
-    public BuildResult perform(final AbstractBuild<?, ?> build, final PluginLogger logger) throws InterruptedException, IOException {
+    public BuildResult perform(final Run<?, ?> build, final FilePath workspace, final PluginLogger logger) throws InterruptedException, IOException {
```

Comment:
Again, I think you should add a `@Deprecated` override for these methods. It's hard to imagine someone extending these or calling directly, but you never know.